### ISSUE TITLE
[Feature Request]: Request Key Exchange with user known Node Hex ID

### ISSUE THREAD

**Issue by Coopersmith-24601** [meshtastic/firmware#4914]
### Platform

Cross-Platform

### Description

There are times, notably when updating firmware or other node db wipes, when User A needs to be able to contact another User B.  However, User B may not be in User A's node db yet.  User B may also have a relatively long period set for broadcasting meaning it may not show up for an extended time.

It would be useful if Node A could force a broadcast _and key exchange_ of Node B by entering in Node B's specific Hex ID.

**Comment by caveman99** [comment]
If node A does not have the key for B it will send the DM 'old style'. Also if Node B can not decrypt a DM becasue it has the wrong key, it sends its own nodeinfo (including its key) asking for reply.

**Comment by Coopersmith-24601** [comment]
Right, but you CAN'T send a DM to a node which isn't in the Node DB yet.

What I'm suggesting is a user be able to input the HEX ID of a node they know which then also pings the node and requests it do a key exchange so users don't have to wait.

**Comment by Xaositek** [comment]
In the apps, you can now do a "Share Contact QR" for any node you have to give it to another user. This is likely the best way to handle this for now.

**Comment by thebentern** [comment]
@jp-bennett is also working on a new user facing key exchange flow which will handle this as well.
